code/drasil-code/lib/Language/Drasil/Choices.hs

```diff
@@ -250,48 +266,77 @@ hasSampleInput [] = False
+makeConstraints = Constraints
+
+-- | Constraint behaviour options within program.
+data ConstraintBehaviour = Warning
+                         | Exception
+
+-- | Renders options for program implementation.
+instance RenderChoices ConstraintBehaviour where
+  showChs Warning = S "Warning"
+  showChs Exception = S "Exception"
+
+-- | External Library Options
+newtype ExtLib = Math ODE
+
+-- | All Information needed to solve an ODE 
+data ODE = ODE{
+  -- FIXME: ODEInfos should be automatically built from Instance models when 
+  -- needed, but we can't do that yet so I'm passing it through Choices instead.
+  -- This choice should really just be for an ODEMethod
+  -- | ODE information.
+  odeInfo :: [ODEInfo],
+  -- | Preferentially-ordered list ODE libraries to try.
+  odeLib :: [ODELibPckg]
+}
+-- | Constructor to create an ODE
+makeODE :: [ODEInfo] -> [ODELibPckg] -> ODE
+makeODE = ODE
+
 -- | Default choices to be used as the base from which design specifications 
 -- can be built.
 defaultChoices :: Choices
 defaultChoices = Choices {
   lang = [Python],
-  modularity = Modular Combined,
-  impType = Program,
-  logFile = "log.txt",
-  logging = [],
-  comments = [],
-  doxVerbosity = Verbose,
-  dates = Hide,
-  onSfwrConstraint = Exception,
-  onPhysConstraint = Warning,
-  inputStructure = Bundled,
-  constStructure = Inline,
-  constRepr = Const,
-  conceptMatch = matchConcepts ([] :: [(SimpleQDef, [CodeConcept])]),
-  spaceMatch = spaceToCodeType, 
-  auxFiles = [ReadME],
-  odeLib = [],
-  odes = []
+  architecture = makeArchit (Modular Combined) Program,
+  dataInfo = makeData Bundled Inline Const,
+  maps = makeMaps 
+    (matchConcepts ([] :: [(SimpleQDef, [CodeConcept])])) 
+    spaceToCodeType,
+  optFeats = makeOptFeats 
+    (makeDocConfig [] Verbose Hide) 
+    (makeLogConfig [] "log.txt") 
+    [ReadME],
+  srsConstraints = makeConstraints Exception Warning,
+  extLibs = []
 }
 
 -- | Renders 'Choices' as 'Sentence's.
-choicesSent :: Choices -> [Sentence] 
+choicesSent :: Choices -> [Sentence]
 choicesSent chs = map chsFieldSent [
     (S "Languages", foldlSent_ $ map (S . show) $ lang chs)
-  , (S "Modularity", showChs $ modularity chs)
-  , (S "Input Structure", showChs $ inputStructure chs)
-  , (S "Constant Structure", showChs $ constStructure chs)
-  , (S "Constant Representation", showChs $ constRepr chs)
-  , (S "Implementation Type", showChs $ impType chs)
-  , (S "Software Constraint Behaviour", showChs $ onSfwrConstraint chs)
-  , (S "Physical Constraint Behaviour", showChs $ onPhysConstraint chs)
-  , (S "Comments", showChsList $ comments chs)
-  , (S "Dox Verbosity", showChs $ doxVerbosity chs)
-  , (S "Dates", showChs $ dates chs)
-  , (S "Log File Name", S $ logFile chs)
-  , (S "Logging", showChsList $ logging chs)
-  , (S "Auxiliary Files", showChsList $ auxFiles chs)
-  ] 
+  , (S "Modularity", showChs $ modularity $ architecture chs)
```

Comment:
Maybe like up into 2 columns? That will make it visually easier to parse what's going on here.

Reply:
Here is what I changed for this part. Is that what we want?
```haskell
    (S "Languages", 
      foldlSent_ $ map (S . show) $ lang chs), 
    (S "Modularity", 
      showChs $ modularity $ architecture chs), 
```

Reply:
No, I meant 2 physical columns on the same line, so the above should look like
```haskell
(S "Languages",       foldlSent_ $ map (S . show) $ lang chs), 
(S "Modularity",      showChs $ modularity $ architecture chs), 
(S "Input Structure", showChs $ inputStructure chs)
```
where I used the next "longer" line for better illustration.

Reply:
Changed the format to 2 physical columns on the same line.